### PR TITLE
[libbeat] kafka message support headers feature

### PR DESCRIPTION
**Pull request by herbxu**
## What does this PR do?
kafka message support headers feature

## Why is it important?
Kafka Message Header lets consumers process message metadata without having to deserialize the message value/payload. This could be useful in the routing of message use case.

## Checklist
- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have made corresponding change to the default configuration files
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## Author's Checklist
xiangwei.xxw@alibaba-inc.com

- [ ]

## Related issues

-

### CODE REVIEW

File: libbeat/outputs/kafka/client.go [hunk 1]
```diff
@@ -76,7 +78,8 @@ func newKafkaClient(
 	index string,
 	key *fmtstr.EventFormatString,
 	topic outil.Selector,
-	writer codec.Codec,
+	headers map[string]string,
+    writer codec.Codec,
```
Comment (rdner) [inline libbeat/outputs/kafka/client.go:82]: could you format the whitespace, please?

**Review by rdner — COMMENTED** [review]
Thank you for your contribution, @herbxu!

In order to merge this PR, we would need a test confirming the new functionality works (see https://github.com/elastic/beats/blob/main/libbeat/outputs/kafka/kafka_integration_test.go). Once it's there, I'm happy to merge the PR.

File: libbeat/outputs/kafka/client.go (hunk 1 above)
Comment (herbxu) [reply libbeat/outputs/kafka/client.go]: I have formatted this code, thanks !

Comment (herbxu) [reply libbeat/outputs/kafka/client.go]: > could you format the whitespace, please?

I have formatted this code, thanks !

**Comment by herbxu** [comment]
> Thank you for your contribution, @herbxu!
> 
> In order to merge this PR, we would need a test confirming the new functionality works (see https://github.com/elastic/beats/blob/main/libbeat/outputs/kafka/kafka_integration_test.go). Once it's there, I'm happy to merge the PR.

I have added the integration test for this feature, thanks !

**Review by rdner — COMMENTED** [review]
Thank you for addressing the comments and adding the test case. I noticed another small improvement that you might want to apply.

It looks like you have not signed the CLA https://www.elastic.co/contributor-agreement

Since you use the privacy feature on GitHub and your commits have a different email address, the automation is not able to detect the CLA signature. However, I checked manually in our contributor database and I could find neither your nickname `herbxu` nor your email address xiangwei.xxw@alibaba-inc.com

After the CLA is signed I'll run the CI pipeline and we can merge this PR.

**Comment by herbxu** [comment]
> 



> Thank you for addressing the comments and adding the test case. I noticed another small improvement that you might want to apply.
> 
> It looks like you have not signed the CLA https://www.elastic.co/contributor-agreement
> 
> Since you use the privacy feature on GitHub and your commits have a different email address, the automation is not able to detect the CLA signature. However, I checked manually in our contributor database and I could find neither your nickname `herbxu` nor your email address [xiangwei.xxw@alibaba-inc.com](mailto:xiangwei.xxw@alibaba-inc.com)
> 
> After the CLA is signed I'll run the CI pipeline and we can merge this PR.

I have signed with the email [xiangwei.xxw@alibaba-inc.com](mailto:xiangwei.xxw@alibaba-inc.com), thanks for your suggestion.

**Comment by rdner** [comment]
Jenkins, please test this.

**Comment by rdner** [comment]
@herbxu just FYI, I had to modify the configuration structure a little bit, see https://github.com/elastic/beats/pull/30369

**Pull request merged**